packages/flutter/lib/src/rendering/grid.dart

```diff
@@ -435,8 +473,8 @@ class RenderGrid extends RenderVirtualViewport<GridParentData> {
       double tileTop = _specification.rowOffsets[placement.row] + gridTopPadding;
       double tileBottom = _specification.rowOffsets[placement.row + placement.rowSpan] + gridTopPadding;
 
-      double childWidth = tileRight - tileLeft - placement.padding.horizontal;
-      double childHeight = tileBottom - tileTop - placement.padding.vertical;
+      double childWidth = math.max(0.0, tileRight - tileLeft - _specification.columnSpacing);
```

Comment:
The GridSpecification's tileWidth and tileHeight values include the row and column spacing, even if there's only one column. This makes things work out nicely but it's a little confusing. I've added some doc to GridSpecification that hopefully clarifies things.